### PR TITLE
feat: Add Bigtable module

### LINKED ISSUE #1045: [Enhancement]: Add Bigtable emulator testcontainer
### Problem

Bigtable is a gcp key/value database that can be used to 

### Solution

Add new project Testcontainers.Bigtable

### Benefit

Provides bigtable testcontainer to those who need it

### Alternatives

People will have to Define the setup for Bigtable themselves.

### Would you like to help contributing this enhancement?

Yes

### PR DESCRIPTION
**Pull request by NorthernLightsDevel**
Add Bigtable emulator container

### CODE REVIEW

**Comment by NorthernLightsDevel** [comment]
Trying to force update to check if tests failed due to error on my part, or some issue elsewhere. No report of failure found.

**Comment by NorthernLightsDevel** [comment]
@HofmeisterAn I believe this is ready for review.
Using the BigtableTableAdminClient to test that it works as expected by creating and retrieving a table.

File: src/Testcontainers.Bigtable/BigtableContainer.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+namespace Testcontainers.Bigtable;
+
+/// <inheritdoc cref="DockerContainer"/>
+[PublicAPI]
+public class BigtableContainer: DockerContainer
+{
+  public readonly string ProjectId = "project-id";
```
Comment (NorthernLightsDevel) [inline src/Testcontainers.Bigtable/BigtableContainer.cs]: Not sure if this is good practice in this repo, And not sure if the specific project-/instance-id values are required by the emulator or not.

File: docs/modules/index.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ await moduleNameContainer.StartAsync();
 | Azure Cosmos DB | `mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator:latest`     | [NuGet](https://www.nuget.org/packages/Testcontainers.CosmosDb)      | [Source](https://github.com/testcontainers/testcontainers-dotnet/tree/develop/src/Testcontainers.CosmosDb)      |
 | Azure SQL Edge  | `mcr.microsoft.com/azure-sql-edge:1.0.7`                            | [NuGet](https://www.nuget.org/packages/Testcontainers.SqlEdge)       | [Source](https://github.com/testcontainers/testcontainers-dotnet/tree/develop/src/Testcontainers.SqlEdge)       |
 | Azurite         | `mcr.microsoft.com/azure-storage/azurite:3.24.0`                    | [NuGet](https://www.nuget.org/packages/Testcontainers.Azurite)       | [Source](https://github.com/testcontainers/testcontainers-dotnet/tree/develop/src/Testcontainers.Azurite)       |
+| Bigtable        | `gcr.io/google.com/cloudsdktool/google-cloud-cli:latest`            | [NuGet](https://www.nuget.org/packages/Testcontainers.Bigtable)      | [Source](https://github.com/testcontainers/testcontainers-dotnet/tree/develop/src/Testcontainers.Bigtable)      |
```
Comment (NorthernLightsDevel) [inline docs/modules/index.md]: Assuming the module should be listed When merged so users can find it.

**Comment by HofmeisterAn** [comment]
> @HofmeisterAn I believe this is ready for review. Using the BigtableTableAdminClient to test that it works as expected by creating and retrieving a table.

Thank you for the PR. I'll review it within the next few days as soon as I have some free time 🕐. I've already taken a brief look, and it seems good so far.

File: src/Testcontainers.Bigtable/BigtableContainer.cs (hunk 1 above)
Comment (HofmeisterAn) [reply src/Testcontainers.Bigtable/BigtableContainer.cs]: I need to take a closer look at it. AFAIK, we do not have something like this in any of the other Google Cloud modules. We simply create the values in the [tests](https://github.com/testcontainers/testcontainers-dotnet/blob/46c05d7abdcc92bf468947ea6b2826dfb36b1908/tests/Testcontainers.PubSub.Tests/PubSubContainerTests.cs#L19). Maybe that will help already.

File: src/Testcontainers.Bigtable/BigtableBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
+namespace Testcontainers.Bigtable;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public class BigtableBuilder: ContainerBuilder<BigtableBuilder, BigtableContainer, BigtableConfiguration>
+{
+  public const string GCloudCliDockerImage = "gcr.io/google.com/cloudsdktool/google-cloud-cli";
```
Comment (HofmeisterAn) [inline src/Testcontainers.Bigtable/BigtableBuilder.cs]:
```suggestion
  public const string GCloudCliDockerImage = "gcr.io/google.com/cloudsdktool/google-cloud-cli:446.0.1-emulators";
```

Can we pin the image to this version? Then, we do not need to pull different versions and can reuse what the other modules are using.

File: src/Testcontainers.Bigtable/BigtableBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+  public const ushort BigtablePort = 9000;
+
+  public const string CMD = "gcloud beta emulators bigtable start --host-port 0.0.0.0:9000";
```
Comment (HofmeisterAn) [inline src/Testcontainers.Bigtable/BigtableBuilder.cs]: Can we pass the command as an exec form to the builder configuration? Similar to [PubSub](https://github.com/testcontainers/testcontainers-dotnet/blob/2229e964cf0ed5b0b5daeac5d8eca5fafdb894fd/src/Testcontainers.PubSub/PubSubBuilder.cs#L46-L47)?

**Review by HofmeisterAn — CHANGES_REQUESTED** [review]

File: src/Testcontainers.Bigtable/BigtableBuilder.cs (hunk 3 above)
Comment (NorthernLightsDevel) [reply src/Testcontainers.Bigtable/BigtableBuilder.cs]: Agreed, I also updated documentation in the index under modules to reflect the change

File: src/Testcontainers.Bigtable/BigtableBuilder.cs (hunk 4 above)
Comment (NorthernLightsDevel) [reply src/Testcontainers.Bigtable/BigtableBuilder.cs]: Done. Didn't consider that this was an option.

File: src/Testcontainers.Bigtable/BigtableContainer.cs (hunk 1 above)
Comment (NorthernLightsDevel) [reply src/Testcontainers.Bigtable/BigtableContainer.cs]: I think that sounds like a fairt option, looks like the Java implementation does not expose these, so removed them to make it more equal to the BigtableEmulatorContainer in java gcloud module.

**Review by HofmeisterAn — APPROVED** [review]
Thank you for the contribution, pull request and for addressing the change requests. PR looks good 🙏.

**Comment by NorthernLightsDevel** [comment]
> Thank you for the contribution, pull request and for addressing the change requests. PR looks good 🙏.

Thanks, will the Bigtable module be available upon next release, and when approximately can I expect it to be available?

**Comment by HofmeisterAn** [comment]
Yes, it will be included in the next release. There is no ETA yet (I do not think it is urgent, you can always fall back on the generic builder). I would like to include a couple more features; having a preview of the reusable API would be awesome.

**Comment by NorthernLightsDevel** [comment]
No problem, I've used copyed the implementation into the project where I needed it anyway, just want to throw the code out to reduce code lines required to maintain within the projects.

**Pull request merged**